File: FragmentLambdaJava/app/src/main/java/com/example/fragmentlambdajava/MainActivity.java
package com.example.fragmentlambdajava;

import android.os.Bundle;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentTransaction;

public class MainActivity extends AppCompatActivity {

    private TextView txtMensaje;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        txtMensaje = findViewById(R.id.txtMensaje);

        RegisterFragment registerFragment = new RegisterFragment();
        registerFragment.setOnRegisterListener(user -> {
            String userInfo = "Nombre: " + user.getFirstname() + " " + user.getLastname() +
                    "\nCorreo: " + user.getEmail() +
                    "\nTeléfono: " + user.getPhone() +
                    "\nUsuario: " + user.getUsername();
            txtMensaje.setText(userInfo);
        });

        FragmentManager fragmentManager = getSupportFragmentManager();
        FragmentTransaction fragmentTransaction = fragmentManager.beginTransaction();
        fragmentTransaction.replace(R.id.frameLayout, registerFragment);
        fragmentTransaction.commit();
    }
}

File: app/src/main/java/com/quantumsoft/laboratorio04/MainActivity.kt
package com.quantumsoft.laboratorio04

import android.os.Bundle
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.commit

class MainActivity : AppCompatActivity() {
    private lateinit var txtMensaje:TextView

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        txtMensaje=findViewById<TextView>(R.id.txtMensaje)
        val blankFragment=BlankFragment.newInstance(response)
        supportFragmentManager.commit {
            setReorderingAllowed(true)
            replace(R.id.frameLayout, blankFragment)
        }
    }


    val response:(String)->Unit={mensaje->
        txtMensaje.text=mensaje

    }

}